[BUG] Failure when loading SparkExecutionEngineConfigClusterSetting

**What is the bug?**
It throws exception when loading SparkExecutionEngineConfigClusterSetting

```
java.lang.RuntimeException: Unable to invoke no-args constructor for class org.opensearch.sql.spark.config.SparkExecutionEngineConfigClusterSetting. Registering an InstanceCreator with Gson for this type may fix this problem.
	at com.google.gson.internal.ConstructorConstructor$14.construct(ConstructorConstructor.java:228) ~[gson-2.8.9.jar:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:212) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:963) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:928) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:877) ~[gson-2.8.9.jar:?]
	at com.google.gson.Gson.fromJson(Gson.java:848) ~[gson-2.8.9.jar:?]
	at org.opensearch.sql.spark.config.SparkExecutionEngineConfigClusterSetting.toSparkExecutionEngineConfig(SparkExecutionEngineConfigClusterSetting.java:34) ~[async-query-2.16.0.0-SNAPSHOT.jar:?]
	at org.opensearch.sql.spark.config.SparkExecutionEngineConfigClusterSettingLoader.lambda$load$0(SparkExecutionEngineConfigClusterSettingLoader.java:30) ~[async-query-2.16.0.0-SNAPSHOT.jar:?]
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:319) ~[?:?]
	at org.opensearch.sql.spark.config.SparkExecutionEngineConfigClusterSettingLoader.load(SparkExecutionEngineConfigClusterSettingLoader.java:27) ~[async-query-2.16.0.0-SNAPSHOT.jar:?]
	at org.opensearch.sql.spark.config.SparkExecutionEngineConfigSupplierImpl.getBuilderFromSettingsIfAvailable(SparkExecutionEngineConfigSupplierImpl.java:31) ~[async-query-2.16.0.0-SNAPSHOT.jar:?]
	at org.opensearch.sql.spark.config.SparkExecutionEngineConfigSupplierImpl.getSparkExecutionEngineConfig(SparkExecutionEngineConfigSupplierImpl.java:25) ~[async-query-2.16.0.0-SNAPSHOT.jar:?]
	at org.opensearch.sql.spark.asyncquery.AsyncQueryExecutorServiceImpl.createAsyncQuery(AsyncQueryExecutorServiceImpl.java:43) ~[async-query-core-2.16.0.0-SNAPSHOT.jar:?]
	at org.opensearch.sql.spark.transport.TransportCreateAsyncQueryRequestAction.doExecute(TransportCreateAsyncQueryRequestAction.java:68) [async-query-2.16.0.0-SNAPSHOT.jar:?]
	at org.opensearch.sql.spark.transport.TransportCreateAsyncQueryRequestAction.doExecute(TransportCreateAsyncQueryRequestAction.java:29) [async-query-2.16.0.0-SNAPSHOT.jar:?]
	at org.opensearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:220) [opensearch-2.16.0.jar:2.16.0]
	at org.opensearch.indexmanagement.rollup.actionfilter.FieldCapsFilter.apply(FieldCapsFilter.kt:118) [opensearch-index-management-2.16.0.0.jar:2.16.0.0]
	at org.opensearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:218) [opensearch-2.16.0.jar:2.16.0]
	at org.opensearch.indexmanagement.controlcenter.notification.filter.IndexOperationActionFilter.apply(IndexOperationActionFilter.kt:39) [opensearch-index-management-2.16.0.0.jar:2.16.0.0]
	at org.opensearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:218) [opensearch-2.16.0.jar:2.16.0]
	at org.opensearch.performanceanalyzer.action.PerformanceAnalyzerActionFilter.apply(PerformanceAnalyzerActionFilter.java:77) [opensearch-performance-analyzer-2.16.0.0.jar:2.16.0.0]
	at org.opensearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:218) [opensearch-2.16.0.jar:2.16.0]
	at org.opensearch.action.support.TransportAction.execute(TransportAction.java:190) [opensearch-2.16.0.jar:2.16.0]
	at org.opensearch.action.support.TransportAction.execute(TransportAction.java:109) [opensearch-2.16.0.jar:2.16.0]
	at org.opensearch.client.node.NodeClient.executeLocally(NodeClient.java:112) [opensearch-2.16.0.jar:2.16.0]
	at org.opensearch.client.node.NodeClient.doExecute(NodeClient.java:99) [opensearch-2.16.0.jar:2.16.0]
	at org.opensearch.client.support.AbstractClient.execute(AbstractClient.java:476) [opensearch-2.16.0.jar:2.16.0]
	at org.opensearch.sql.spark.rest.RestAsyncQueryManagementAction.lambda$executePostRequest$1(RestAsyncQueryManagementAction.java:137) [async-query-2.16.0.0-SNAPSHOT.jar:?]
	at org.opensearch.sql.datasources.utils.Scheduler.lambda$withCurrentContext$0(Scheduler.java:30) [datasources-2.16.0.0-SNAPSHOT.jar:?]
	at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:882) [opensearch-2.16.0.jar:2.16.0]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) [?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) [?:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
Caused by: java.lang.UnsupportedOperationException: Cannot allocate class org.opensearch.sql.spark.config.SparkExecutionEngineConfigClusterSetting
	at com.google.gson.internal.UnsafeAllocator$4.newInstance(UnsafeAllocator.java:104) ~[gson-2.8.9.jar:?]
	at com.google.gson.internal.ConstructorConstructor$14.construct(ConstructorConstructor.java:225) ~[gson-2.8.9.jar:?]
	... 32 more
```


**How can one reproduce the bug?**

1. With 2.16.0, add "plugins.query.executionengine.spark.config" config in config/opensearch.yml, and start
2. send any async_query request
3. It returns error

**What is your host/environment?**
 - OS: [e.g. iOS] Linux 6.1.79-99.167.amzn2023.x86_64
 - Version [e.g. 22] OpenSearch 2.16.0
 - Plugins

**Do you have any screenshots?**
If applicable, add screenshots to help explain your problem.

**Do you have any additional context?**
Add any other context about the problem.

--- a/async-query/src/main/java/org/opensearch/sql/spark/config/SparkExecutionEngineConfigClusterSetting.java
+++ b/async-query/src/main/java/org/opensearch/sql/spark/config/SparkExecutionEngineConfigClusterSetting.java
@@ -7,8 +7,10 @@ package org.opensearch.sql.spark.config;
 
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.google.gson.Gson;
+import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.Data;
+import lombok.NoArgsConstructor;
 
 /**
  * This POJO is just for reading stringified json in `plugins.query.executionengine.spark.config`
@@ -16,6 +18,8 @@ import lombok.Data;
  */
 @Data
 @Builder
+@AllArgsConstructor
+@NoArgsConstructor
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class SparkExecutionEngineConfigClusterSetting {
   // optional

